### PR TITLE
Improvements to shared/ static module builds and debugging

### PR DESCRIPTION
**Pull request by Shillaker**
This moves a lot of the compiler flags and configuration around building shared libraries upstream into this repo. 

This avoids such information being spread over multiple files, but it's still duplicated in this project and the `faasm-toolchain`. Another piece of work will be to unify the two and move all this config into `faasm-toolchain`.

Part of #302

### CODE REVIEW

File: faasmcli/faasmcli/tasks/libs.py [hunk 1]
```diff
@@ -227,33 +227,3 @@ def eigen(ctx, verbose=False):
         check=True,
     )
 
-
```
Comment (Shillaker) [inline faasmcli/faasmcli/tasks/libs.py:230]: The BLAS stuff has been moved to the `faasm-clapack` submodule

File: func/dynlink/CMakeLists.txt [hunk 2]
```diff
@@ -5,36 +5,5 @@ set(BUILD_SHARED_LIBS ON)
 
 include_directories(include)
 
-# This is critical otherwise the linker tries to statically link in standard
```
Comment (Shillaker) [inline func/dynlink/CMakeLists.txt:8]: This configuration is all now handled by the toolchain

File: src/wavm/codegen.cpp [hunk 3]
```diff
@@ -6,13 +6,17 @@
...
 #include <WAVM/WASTParse/WASTParse.h>
 
+#include <faabric/util/logging.h>
 #include <faabric/util/files.h>
 
 using namespace WAVM;
 
 namespace wasm {
-std::vector<uint8_t> wavmCodegen(std::vector<uint8_t>& bytes)
+std::vector<uint8_t> wavmCodegen(std::vector<uint8_t>& bytes,
```
Comment (csegarragonz) [inline src/wavm/codegen.cpp:15]: So these changes to `doCodegen` and `wavmCodegen` are for better error reporting I guess? It's alright, just a bit confusing that they are included in this PR.

**Review by csegarragonz — APPROVED** [review]

**Comment by csegarragonz** [comment]
LGTM

File: src/wavm/codegen.cpp (hunk 3 above)
Comment (Shillaker) [reply src/wavm/codegen.cpp:15]: I guess it depends on how strict we want to be with single purpose PRs. Given how long the build takes at the moment I think it's OK to include small improvements like this alongside other changes.

Comment (Shillaker) [reply src/wavm/codegen.cpp:15]: I take your point that the title is kind of misleading. Although it's updating the submodules that contain changes related to Horovod/ MXNet, the changes in this repo are more generally relevant to building libraries. I've renamed the PR to better reflect the fact that it's a bit of a catch-all

**Pull request merged**